feat: diode-proto - set explicit fields presence

mfiedorowicz opened the pull request:
This is to track protobuf fields / detect if field has value (provided vs default zero-value), which will allow us to reset fields in NetBox models if instructed by providing zero-values, i.e. an interface with `enabled` field with `false` value to disable interface (currently does nothing it interface is enabled), similarly with any other optional field (an interface's `description` with `Lorem ipsum` cannot be reset to an empty string, while these changes will allow to provide `description=""`  to reset in the NetBox).

ref: https://protobuf.dev/programming-guides/field_presence/#presence-in-proto3-apis

Bonus: `make` recipes for development docker compose (building image instead of using `latest` from container registry) and don't build image with `make docker-compose-up` (pulls `latest` tags from container registry).

![Test Execution Report](https://github.com/user-attachments/assets/b2398a1d-10a5-4850-b748-1d652dfc40aa)

mfiedorowicz committed:
chore: docker - various adjustments

- add PROJECT_NAME variable (default: diode)
- add DIODE_TAG variable (default: latest)
- add docker-compose-dev-up and docker-compose-dev-down make recipes

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
chore: docker - various adjustments

- add PROJECT_NAME variable (default: diode)
- add DIODE_TAG variable (default: latest)
- add docker-compose-dev-up and docker-compose-dev-down make recipes

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
feat: diode-proto - set explicit fields presence

this change is to track protobuf fields / detect if field has value (provided vs default zero-value)

ref: https://protobuf.dev/programming-guides/field_presence/#presence-in-proto3-apis
Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
feat: diode-proto - set explicit fields presence

this change is to track protobuf fields / detect if field has value (provided vs default zero-value)

ref: https://protobuf.dev/programming-guides/field_presence/#presence-in-proto3-apis
Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

leoparente reviewed (approved):
LGTM. Nice adding on compose for developers

mfiedorowicz committed:
feat: diode-proto - update docs

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
feat: diode-proto - update docs

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
add integration tests to assert explicit field presence functionality

testing:
- device: change description to an empty string
- interface: disable previously enabled interface

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>

mfiedorowicz committed:
Merge remote-tracking branch 'origin/fix/OBS-490-proto-explicit-field-presence' into fix/OBS-490-proto-explicit-field-presence

mfiedorowicz committed:
Merge branch 'refs/heads/develop' into fix/OBS-490-proto-explicit-field-presence

mfiedorowicz committed:
Merge branch 'refs/heads/develop' into fix/OBS-490-proto-explicit-field-presence

mfiedorowicz committed:
missing new line at EOF

Signed-off-by: Michal Fiedorowicz <mfiedorowicz@netboxlabs.com>
